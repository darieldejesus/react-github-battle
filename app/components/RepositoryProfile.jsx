import React, { Component } from 'react';
import PropTypes from 'prop-types';

/**
 * Populate the main information about the repos.
 */
const RepositoryProfile = props => {
  const { position, image, name, url, login, stars } = props;
  return (
    <div className='popular-item'>
      <h5 className='popular-rank'>#{position + 1}</h5>
      <img 
        className='avatar' 
        src={image}
        alt={`Avatar for ${name}`} />
      <p><a href={url}>{name}</a></p>
      <p>@{login}</p>
      <p>{stars} stars</p>
    </div>
  )
}

RepositoryProfile.propTypes = {
  image: PropTypes.string.isRequired,
  login: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  position: PropTypes.number.isRequired,
  stars: PropTypes.number.isRequired,
  url: PropTypes.string.isRequired
}

export default RepositoryProfile;
