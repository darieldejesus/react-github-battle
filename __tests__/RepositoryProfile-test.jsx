import React from 'react';
import { shallow } from 'enzyme';

import RepositoryProfile from '../app/components/RepositoryProfile';

describe('<RepositoryProfile />', () => {
  it('Repository profile is rendered correctly.', () => {
    const props = {
      image: 'https://example.com/image.jpg',
      login: 'profilename',
      name: 'Profile Name',
      position: 1,
      stars: 10,
      url: 'https://example.com/profilename'
    };
    const repositoryProfile = shallow(
      <RepositoryProfile {...props} />
    );
    expect(repositoryProfile).toMatchSnapshot();
  });
});
