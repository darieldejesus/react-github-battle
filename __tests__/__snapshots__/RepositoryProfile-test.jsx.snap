// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RepositoryProfile /> Repository profile is rendered correctly. 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="popular-item"
  >
    <h5
      className="popular-rank"
    >
      #
      2
    </h5>
    <img
      alt="Avatar for Profile Name"
      className="avatar"
      src="https://example.com/image.jpg"
    />
    <p>
      <a
        href="https://example.com/profilename"
      >
        Profile Name
      </a>
    </p>
    <p>
      @
      profilename
    </p>
    <p>
      10
       stars
    </p>
  </div>,
  "nodes": Array [
    <div
      className="popular-item"
    >
      <h5
        className="popular-rank"
      >
        #
        2
      </h5>
      <img
        alt="Avatar for Profile Name"
        className="avatar"
        src="https://example.com/image.jpg"
      />
      <p>
        <a
          href="https://example.com/profilename"
        >
          Profile Name
        </a>
      </p>
      <p>
        @
        profilename
      </p>
      <p>
        10
         stars
      </p>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <RepositoryProfile
        image="https://example.com/image.jpg"
        login="profilename"
        name="Profile Name"
        position={1}
        stars={10}
        url="https://example.com/profilename"
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "image": "https://example.com/image.jpg",
          "login": "profilename",
          "name": "Profile Name",
          "position": 1,
          "stars": 10,
          "url": "https://example.com/profilename",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="popular-item"
        >
          <h5
            className="popular-rank"
          >
            #
            2
          </h5>
          <img
            alt="Avatar for Profile Name"
            className="avatar"
            src="https://example.com/image.jpg"
          />
          <p>
            <a
              href="https://example.com/profilename"
            >
              Profile Name
            </a>
          </p>
          <p>
            @
            profilename
          </p>
          <p>
            10
             stars
          </p>
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="popular-item"
        >
          <h5
            className="popular-rank"
          >
            #
            2
          </h5>
          <img
            alt="Avatar for Profile Name"
            className="avatar"
            src="https://example.com/image.jpg"
          />
          <p>
            <a
              href="https://example.com/profilename"
            >
              Profile Name
            </a>
          </p>
          <p>
            @
            profilename
          </p>
          <p>
            10
             stars
          </p>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <RepositoryProfile
    image="https://example.com/image.jpg"
    login="profilename"
    name="Profile Name"
    position={1}
    stars={10}
    url="https://example.com/profilename"
  />,
}
`;
